mcs/class/corlib/System/Console.cs

```diff
@@ -141,45 +141,31 @@ static void SetupStreams (Encoding inputEncoding, Encoding outputEncoding)
 		{
 #if !NET_2_1
 			if (!Environment.IsRunningOnWindows && ConsoleDriver.IsConsole) {
-				StreamWriter w = new CStreamWriter (OpenStandardOutput (0), outputEncoding, true);
-				w.AutoFlush = true;
-				stdout = TextWriter.Synchronized (w);
-
-				w = new CStreamWriter (OpenStandardOutput (0), outputEncoding, true);
-				w.AutoFlush = true;
-				stderr = TextWriter.Synchronized (w);
-				
 				stdin = new CStreamReader (OpenStandardInput (0), inputEncoding);
-			} else {
-#endif
-#if MONOTOUCH
-				stdout = new NSLogWriter ();
-#else
-				stdout = new UnexceptionalStreamWriter (OpenStandardOutput (0), outputEncoding);
-				((StreamWriter)stdout).AutoFlush = true;
+				stdout = TextWriter.Synchronized (new CStreamWriter (OpenStandardOutput (0), outputEncoding, true) { AutoFlush = true });
+				stderr = TextWriter.Synchronized (new CStreamWriter (OpenStandardError (0), outputEncoding, true) { AutoFlush = true });
+			} else
 #endif
-				stdout = TextWriter.Synchronized (stdout);
+			{
+				stdin = TextReader.Synchronized (new UnexceptionalStreamReader (OpenStandardInput (0), inputEncoding));
 
+#if MONODROID
+				if (LogcatTextWriter.IsRunningOnAndroid ()) {
+					stdout = TextWriter.Synchronized (new LogcatTextWriter ("mono-stdout", stdout));
+					stderr = TextWriter.Synchronized (new LogcatTextWriter ("mono-stderr", stderr));
```

Comment:
I think this is changing behavior here.
Before in the MONODROID case, stdout/stderr were initialized to a UnexceptionalStreamWriter before being passed to LogcatTextWriter.

Now they're not initialized at this point from what I can see.